server/api/region.go

```diff
@@ -257,6 +258,61 @@ func (h *regionsHandler) GetEmptyRegion(w http.ResponseWriter, r *http.Request)
 	h.rd.JSON(w, http.StatusOK, regionsInfo)
 }
 
+type regionSize struct {
+	Size  int64 `json:"size"`
+	Count int   `json:"count"`
+}
+
+type regionKeys struct {
+	Keys  int64 `json:"keys"`
+	Count int   `json:"count"`
+}
+
+type regionHist struct {
+	SizeHist []*regionSize `json:"region size histogram"`
+	KeysHist []*regionKeys `json:"region keys histogram"`
+}
+
+func (h *regionsHandler) GetHistogram(w http.ResponseWriter, r *http.Request) {
+	rc := getCluster(r.Context())
+	regions := rc.GetRegions()
+	sizeHistgoram := histogram.New(10)
+	keysHistgoram := histogram.New(10)
+	for _, region := range regions {
+		sizeHistgoram.Insert(float64(region.GetApproximateSize()))
+		keysHistgoram.Insert(float64(region.GetApproximateKeys()))
+	}
+	sizeBins := sizeHistgoram.Bins()
+	keysBins := keysHistgoram.Bins()
+	var sizeHist [10]*regionSize
+	var keysHist [10]*regionKeys
+
+	for i, bin := range sizeBins {
+		if bin == nil {
+			continue
+		}
+		reginSize := &regionSize{}
+		reginSize.Count = bin.Count
+		reginSize.Size = int64(bin.Mean())
+		sizeHist[i] = reginSize
+	}
+	for i, bin := range keysBins {
+		if bin == nil {
+			continue
+		}
+		reginKeys := &regionKeys{}
+		reginKeys.Count = bin.Count
+		reginKeys.Keys = int64(bin.Mean())
+		keysHist[i] = reginKeys
+	}
+
+	regionHist := &regionHist{
```

Comment:
do we need to output bounds?

Reply:
I guess you mean the bound of the histogram bin. The histogram implemented in this method is not equally divided. For example, our histogram uses 10 bins and it has 9 bounds. Their bound values are only close, but not equal. . Therefore, no specific bound can be output.

Reply:
yep,I think the approximation is acceptable. Or if we want draw a histogram with bound by pd-ctl, do you have other ideas?

Reply:
I can add a numerical parameter to pd-ctl to indicate that the histogram needs to be divided into several equal parts, for example, it is now hard-coded into 10 equal parts. The default value is 10, and the value range is [1,100]. How about? @lhy1024

Reply:
@lhy1024 I have another question to confirm. Does the output format meet your needs? If not, please give me a style of the output format, I will do the output according to this style, thank you!

Reply:
Good, you are free to design style. I only want to get the `bound`~

Reply:
Yes, we need to actually know the bound, event more we can specify the bound through `pc-ctl`.
The main purpose is that we can use the histogram to determine the parameters of the `region-merge`, such as we know [0,20] have 20w region, we can set:
```
 "max-merge-region-size": 20,
```

Reply:
@nolouch @lhy1024 
OK.Then I provide two hist commands:
pd-ctl check region histsize
pd-ctl check region histkeys

Both commands have input positive integer parameters, that is, bound. The default value of the histsize parameter is 10, and the value range is [1, positive infinity)
The default value of the histkeys parameter is 10000, and the value range is [1000, positive infinity)
The output looks like this:
[0,10): 1000
[10,20): 1000
[20,30): 1000
[30,40): 1000
[40,50): 1000
[50,998]: 23

Note that there are two important points here:
The first is split into two commands;
The second is that bound is entered manually and is limited by default and minimum values.

Reply:
Good, can we set the bound with `pd-ctl`? Or do we need to set the bound? @nolouch

Reply:
The output looks good, better to support to set the bound in `pd-ctl`.  cc @mantuliu

Reply:
yes,support to set the bound in pd-ctl. @nolouch @lhy1024  PTAL.

Reply:
> Thank you very much. would you like to show output now?

bin/pd-ctl region check hist-keys 10000
{
  "[0,10000)": 1000020,
  "[200000,210000)": 50055,
  "[210000,220000)": 49702,
  "[220000,230000)": 50162,
  "[230000,240000)": 50392,
  "[240000,250000)": 49935,
  "[250000,260000)": 49662,
  "[260000,270000)": 50302,
  "[270000,280000)": 49778,
  "[280000,290000)": 50155,
  "[290000,300000)": 49838,
  "[300000,310000)": 49715,
  "[310000,320000)": 50019,
  "[320000,330000)": 50076,
  "[330000,340000)": 50040,
  "[340000,350000)": 50178,
  "[350000,360000)": 49887,
  "[360000,370000)": 49910,
  "[370000,380000)": 50011,
  "[380000,390000)": 49977,
  "[390000,400000)": 50226
}
like this

Reply:
look good to me.

Reply:
I just realized that this format may not conform to the principle of json self-explanation. how about `{"start":0,"end":10000,"count":1000020}` @mantuliu